### PR TITLE
ci: Run more ci configs on cirrus

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK, it seems split from #19179.

Could pr title and/or description mention arm build as well?

File: .cirrus.yml [hunk 1]
```diff
@@ -53,6 +53,14 @@ global_task_template: &GLOBAL_TASK_TEMPLATE
 #  install_script:
 #    - choco install python --version=3.7.7 -y
 
+task:
+  name: 'ARM  [GOAL: install]  [buster]  [unit tests, no functional tests]'
```
Comment (hebasto) [inline .cirrus.yml:57]: faf07014b9c166692ca2bab10695f00a90990027, nit:
```suggestion
  name: 'ARM  [GOAL: install]  [buster]  [unit tests in QEMU, no functional tests]'
```

File: .cirrus.yml [hunk 2]
```diff
@@ -135,3 +135,17 @@ task:
     image: ubuntu:bionic
   env:
     FILE_ENV: "./ci/test/00_setup_env_mac.sh"
+
+task:
+  name: 'macOS 10.14 native [GOAL: install] [GUI] [no depends]'
+  macos_brew_addon_script:
+    - brew install boost libevent berkeley-db4 qt miniupnpc qrencode sqlite ccache zeromq qrencode sqlite libtool automake pkg-config coreutils gnu-getopt
```
Comment (hebasto) [inline .cirrus.yml:142]: fafc0158f99d90078223461631ef2829bd5fcec6

The `qrencode` and `sqlite` packages are mentioned twice :)

Why the `coreutils` and `gnu-getopt` packages are required (comparing to [doc/build-osx.md](https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md))?

**Review by hebasto — CHANGES_REQUESTED** [review]
Approach ACK fafc0158f99d90078223461631ef2829bd5fcec6.

File: ci/test/00_setup_env_mac_host.sh [hunk 3]
```diff
@@ -10,6 +10,7 @@ export HOST=x86_64-apple-darwin16
 export PIP_PACKAGES="zmq"
 export GOAL="install"
 export BITCOIN_CONFIG="--with-gui --enable-reduce-exports --enable-werror --with-boost-process"
+export CI_OS_NAME="osx"
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply ci/test/00_setup_env_mac_host.sh]: thx, changes osx->macos

File: .cirrus.yml (hunk 2 above)
Comment (MarcoFalke) [reply .cirrus.yml]: thanks removed duplicate packets. 

getopt is needed for `IN_GETOPT_BIN="/usr/local/opt/gnu-getopt/bin/getopt"` in the same commit

**Review by hebasto — APPROVED** [review]
ACK 77777c8b5e3d0edbe1178d982a778b7cc9964b70

Does the latest commit 77777c8b5e3d0edbe1178d982a778b7cc9964b70 keeps compatibility with Travis CI?

**Comment by MarcoFalke** [comment]
Yes, see https://travis-ci.org/github/bitcoin/bitcoin/builds/742437881

**Pull request merged**